### PR TITLE
Add parsing of doc comments

### LINKED ISSUE #147: Separate regular comments from Doc Strings
It would be great if documentation comments made with `///`, `//!`, etc. were matched to different nodes than normal comments.  I'd like to be able treat them separately when doing syntax highlighting.

### PR DESCRIPTION
**Pull request by Jesse-Bakker**
Fixes #147

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -39,13 +39,15 @@ const primitive_types = numeric_types.concat(['bool', 'str', 'char'])
 module.exports = grammar({
   name: 'rust',
 
-  extras: $ => [/\s/, $.line_comment, $.block_comment],
+  extras: $ => [/\s/, $.line_doc_comment, $.line_comment, $.block_comment, $.block_doc_comment],
```
Comment (the-mikedavis) [inline grammar.js:42]: I think it should be possible to expose doc contents without introducing new named nodes for doc comments by making the `doc_content` nodes optional fields on `line_comment` and `block_comment`. For example, line comments could be structured like this:

```
// Line comment
/// Doc comment

---

(source
  (line_comment)
  (line_comment
    docs: (doc_content)))
```

Then the query for injecting markdown only needs to match against the `doc_content` node and highlights queries won't need any changes.

Comment (nickeb96) [reply grammar.js:42]: It's better to make these two separate nodes.  In this repository's query file we would just link `line_comment`/`block_comment` to `@comment` and then `line_doc_comment`/`block_doc_comment` to `@comment.doc`.

Anything marked as `@comment.doc` will automatically use `@comment` as a fallback so anything using the queries (like colors schemes) won't need any changes in the first place.

Comment (the-mikedavis) [reply grammar.js:42]: You could still capture these with separate `comment` and `comment.doc` captures by matching on the presence or absence of the `docs` field:

```scm
(line_comment !docs) @comment
(line_comment docs: _) @comment.doc
; same for block comments
```

Comment (359013996) [reply grammar.js:42]: chart

**Comment by nickeb96** [comment]
@maxbrunsfeld could you please review or merge this?

**Comment by clason** [comment]
superseded by #195

**Pull request closed without merging**